src/content/docs/en/reference/loader-reference.mdx

```diff
@@ -3,6 +3,450 @@ title: Astro Loader API
+    for (const item of feed.items) {
+      const data = await parseData({
+        id: item.guid,
+        data: item,
+      });
+      const digest = generateDigest(data);
+      store.set({
+        id,
+        data,
+        rendered: {
+          html: data.description ?? "",
+        },
+        digest,
+      });
+    }
+```
+
+#### `entries`
+
+<p>
+
+**Type**: `() => Array<[id: string, DataEntry]>`
+</p>
+
+Get all entries in the collection as an array of key-value pairs.
+
+#### `keys`
+**Type**: `() => Array<string>`
+
+Get all the keys of the entries in the collection.
+
+#### `values`
+
+<p>
+
+**Type**: `() => Array<DataEntry>`
+</p>
+
+Get all entries in the collection as an array.
+
+#### `delete`
+
+<p>
+
+**Type**: `(key: string) => void`
+</p>
+
+Delete an entry from the store by its ID.
+
+#### `clear`
+
+<p>
+
+**Type**: `() => void`
+</p>
+
+Clear all entries from the collection.
+
+#### `has`
+
+<p>
+
+**Type**: `(key: string) => boolean`
+</p>
+
+Check if an entry exists in the store by its ID.
+
+### `DataEntry`
+
+This is the type of the object that is stored in the data store. It has the following properties:
+
+#### `id`
+
+<p>
+
+**Type**: `string`
+</p>
+
+An identifier for the entry, which must be unique within the collection. This is used to look up the entry in the store, and is the key used with `getEntry` for that collection. 
+
+#### `data`
+
+<p>
+
+**Type**: `Record<string, unknown>`
+</p>
+
+The actual data for the entry. When a user accesses the collection, this will have TypeScript types generated according to the collection schema. It the responsibility of the loader to use [`parseData`](#parsedata) to validate and parse the data before storing it in the data store: no validation is done when getting or setting the data.
+
+#### `filePath`
+
+<p>
+
+**Type**: `string | undefined`
+</p>
+
+A path to the file that is the source of this entry, relative to the root of the site. This only applies to file-based loaders, and is used to resolve paths such as images or other assets. If not set, then any fields in the schema that use [the `image()` helper](/en/guides/images/#images-in-content-collections) will be treated as [public paths](/en/guides/images/#where-to-store-images) and not transformed.
+
+#### `body`
+
+<p>
+
+**Type**: `string | undefined`
+</p>
+
+The raw body of the entry, if applicable. If the entry includes [rendered content](#rendered), then this field can be used to store the raw source. This is optional, and is not used internally.
+
+#### `digest`
+
+<p>
+
+**Type**: `string | undefined`
+</p>
+
+An optional content digest for the entry. This can be used to check if the data has changed. If the digest is passed when setting an entry, it will only update the entry if the digest does not match the existing entry. The format of the digest is up to the loader, but it should be a string that changes when the data changes. The [`generateDigest`](#generatedigest) function can be used for this.
```

Comment:
```suggestion
An optional content digest for the entry. This can be used to check if the data has changed.

When [setting an entry](#set), the entry will only update if the digest does not match an existing entry with the same ID.

The format of the digest is up to the loader, but it must be a string that changes when the data changes. This can be done with the [`generateDigest`](#generatedigest) function.
```
Alright, does this pair well with our `set` entry?

Reply:
Looks good. Do you think it needs to call out the fact that it only does the conditional insert if the digest is provided. If it's unset, it will always update the entry and return true.

Reply:
I don't *think* we need to call that out? The field is optional, and it can't match if it's not provided in the first place? Let's go understated to start, then we'll find out if people are confused and need the call out!